### PR TITLE
obs-qsv11: Fix target usage migration string comparisons

### PR DESCRIPTION
**Pull request by gxalpha**
### Description

Fixes problems with the astrcmpi logic, as well as a typo that said "vertfast" instead "veryfast".
Also makes the method a void because it always returns true and isn't checked anyways.

### Motivation and Context

Was found while investigating https://github.com/obsproject/obs-studio/issues/9574.

### How Has This Been Tested?

Hasn't.

### Types of changes

- Bug fix (non-breaking change which fixes an issue)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [ ] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by RytoEX** [comment]
Was looking into this myself. While these changes are fine, I think the core issue is that `update_targetusage` is only called via `update_settings`, which means that the target usage is not actually updated until that is invoked. If you start an encoder session, then `update_settings` is called, which calls `update_params`, which calls `update_targetusage` and internally sets the appropriate TU but doesn't change the UI. `update_latency`, `update_enhancements`, and `update_ratecontrol` are all called in `rate_control_modified` (those first two are also called in `update_params`). If we call `update_targetusage` from `rate_control_modified` like the others, it works fine in that the UI value is now updated.

File: plugins/obs-qsv11/obs-qsv11.c [hunk 1]
```diff
@@ -328,33 +328,31 @@ static bool update_enhancements(obs_data_t *settings)
...
-static bool update_targetusage(obs_data_t *settings)
+static void update_targetusage(obs_data_t *settings)
 {
 	const char *target_usage =
 		obs_data_get_string(settings, "target_usage");
 
-	if ((astrcmpi(target_usage, "veryslow") ||
-	     astrcmpi(target_usage, "quality")) == 0)
+	if (astrcmpi(target_usage, "veryslow") == 0 ||
+	    astrcmpi(target_usage, "quality") == 0)
```
Comment (tytan652) [inline plugins/obs-qsv11/obs-qsv11.c:337]:
```suggestion
	if ((astrcmpi(target_usage, "veryslow") == 0) ||
	    (astrcmpi(target_usage, "quality") == 0))
```

File: plugins/obs-qsv11/obs-qsv11.c [hunk 2]
```diff
@@ -328,33 +328,31 @@ static bool update_enhancements(obs_data_t *settings)
...
 		obs_data_set_string(settings, "target_usage",
 				    "TU1: Slowest (Best Quality)");
 	else if (astrcmpi(target_usage, "slower") == 0)
 		obs_data_set_string(settings, "target_usage", "TU2: Slower");
 	else if (astrcmpi(target_usage, "slow") == 0)
 		obs_data_set_string(settings, "target_usage", "TU3: Slow");
-	else if ((astrcmpi(target_usage, "medium") ||
-		  astrcmpi(target_usage, "balanced")) == 0)
+	else if (astrcmpi(target_usage, "medium") == 0 ||
+		 astrcmpi(target_usage, "balanced") == 0)
```
Comment (tytan652) [inline plugins/obs-qsv11/obs-qsv11.c:345]:
```suggestion
	else if ((astrcmpi(target_usage, "medium") == 0) ||
		 (astrcmpi(target_usage, "balanced") == 0))
```

File: plugins/obs-qsv11/obs-qsv11.c [hunk 3]
```diff
@@ -328,33 +328,31 @@ static bool update_enhancements(obs_data_t *settings)
...
 		obs_data_set_string(settings, "target_usage",
 				    "TU4: Balanced (Medium Quality)");
 	else if (astrcmpi(target_usage, "fast") == 0)
 		obs_data_set_string(settings, "target_usage", "TU5: Fast");
 	else if (astrcmpi(target_usage, "faster") == 0)
 		obs_data_set_string(settings, "target_usage", "TU6: Faster");
-	else if ((astrcmpi(target_usage, "vertfast") ||
-		  astrcmpi(target_usage, "speed")) == 0)
+	else if (astrcmpi(target_usage, "veryfast") == 0 ||
+		 astrcmpi(target_usage, "speed") == 0)
```
Comment (tytan652) [inline plugins/obs-qsv11/obs-qsv11.c:353]:
```suggestion
	else if ((astrcmpi(target_usage, "veryfast") == 0) ||
		 (astrcmpi(target_usage, "speed") == 0))
```

**Review by tytan652 — CHANGES_REQUESTED** [review]
`(astrcmpi || astrcmpi) == 0` is really bad because the returned 0 is casted as false but if one of the `astrcmpi` returns something different than 0 we get a cast to true and so even if one if them return 0/false the other returns true and so the `||` makes a true from both and is not equal to 0

**Comment by derrod** [comment]
I'm not sure what you're saying, the code is doing `astrcmpi(...) == 0 || astrcmpi(...) == 0`, not `(astrcmpi || astrcmpi) == 0`.

**Comment by tytan652** [comment]
Priorities in this situation are messy, feel free to resolve them.

**Comment by gxalpha** [comment]
@tytan652 For reference, C++ operator precedence can be checked here: https://en.cppreference.com/w/cpp/language/operator_precedence
Equality is group 10, logical OR is group 14, so equality will be evaluated first.

@RytoEX do you want me to add another call to `update_targetusage` to this PR or do you consider that to be standalone? I could unlink the open issue in the latter case.

**Review by tytan652 — APPROVED** [review]
The function looks good to me

**Comment by RytoEX** [comment]
> @tytan652 For reference, C++ operator precedence can be checked here: https://en.cppreference.com/w/cpp/language/operator_precedence Equality is group 10, logical OR is group 14, so equality will be evaluated first.
> 
> @RytoEX do you want me to add another call to `update_targetusage` to this PR or do you consider that to be standalone? I could unlink the open issue in the latter case.

Let's keep the scope small and just fix the obvious errors here. We can handle adding `update_targetusage` to `rate_control_modified` in a separate PR.

**Pull request merged**